@here/harp-examples/src/features_custom.ts

```diff
@@ -0,0 +1,440 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Style, StyleSet } from "@here/harp-datasource-protocol";
+import { FeaturesDataSource, MapViewFeature } from "@here/harp-features-datasource";
+import { GeoCoordinates } from "@here/harp-geoutils";
+import { MapControls, MapControlsUI } from "@here/harp-map-controls";
+import { CopyrightInfo, MapView } from "@here/harp-mapview";
+import { APIFormat, OmvDataSource } from "@here/harp-omv-datasource";
+import * as THREE from "three";
+import { accessToken } from "../config";
+import { COUNTRIES } from "./../resources/countries";
+
+/**
+ * This example illustrates how to add custom features in [[MapView]]. They are handled through a
+ * special kind of [[DataSource]], a [[FeaturesDataSource]].
+ *
+ * First we create a base map. For more details, check the `hello` example.
+ * ```typescript
+ * [[include:harp_demo_features_example_0.ts]]
+ * ```
+ *
+ * Then we generate a custom [[StyleSet]] for the countries, with a color gradient based on the
+ * year that the country joined the EU. The [[Style]]s are all an [[ExtrudedPolygonStyle]] with only
+ * a variation in the color.
+ * ```typescript
+ * [[include:harp_demo_features_example_1.ts]]
+ * ```
+ *
+ * In order to have the extrusion animation for all the individual sets of countries joining, we
+ * group them and handle them in separate datasources, so that the various datasources' tiles will
+ * get animated independently and highlight new territories. Each of these datasources is a
+ * [[FeaturesDataSource]]. The features generation API is on them, but they don't own the feature
+ * yet.
+ * ```typescript
+ * [[include:harp_demo_features_example_2.ts]]
+ * ```
+ *
+ * We then add the feature manually:
+ * ```typescript
+ * [[include:harp_demo_features_example_3.ts]]
+ * ```
+ *
+ * The rest of the code is example-dependent logic to handle the countries sets, with the joining
+ * and leaving mechanism, that adds and removes datasources.
+ */
+export namespace DemoFeaturesExample {
```

Comment:
this is a great example, but it is quite complex, I think it would be good to have a very simple demo which just adds a single geo json object, for those who may not be so advanced.